packages/@cdktf/cli-core/src/lib/models/terraform-cli.ts

```diff
@@ -106,28 +107,51 @@ export class TerraformCli implements Terraform {
         createTerraformLogHandler(phase, filter)(stderr.toString(), true);
   }
 
-  public async init(needsUpgrade: boolean, noColor?: boolean): Promise<void> {
+  public async init(
+    needsUpgrade: boolean,
+    noColor: boolean,
+    migrateState: boolean
+  ): Promise<void> {
     await this.setUserAgent();
 
-    const args = ["init", "-input=false"];
+    const args = ["init"];
     if (needsUpgrade) {
       args.push("-upgrade");
     }
     if (noColor) {
       args.push("-no-color");
     }
 
-    await exec(
-      terraformBinaryName,
-      args,
+    const stdout = this.onStdout("init");
+    const stderr = this.onStderr("init");
+    const { actions, progress } = spawnPty(
       {
-        cwd: this.workdir,
-        env: process.env,
-        signal: this.abortSignal,
+        file: terraformBinaryName,
+        args,
+        options: {
+          cwd: this.workdir,
+          env: process.env as any,
+        },
       },
-      this.onStdout("init"),
-      this.onStderr("init")
+      (data) => {
+        stdout(data);
+        if (data.includes("Should Terraform migrate your existing state?")) {
+          if (migrateState) {
+            actions.writeLine("yes");
+          } else {
+            actions.writeLine("no");
+            stderr(
+              "Please pass the --migrate-state flag to migrate your state"
+            );
```

Comment:
Nice!

Will `terraform init` return with a non-zero exit code if we send "no"? If I remember correctly `terraform apply` does so when we reject the plan.

Reply:
Hmm, I'm unsure. I think I'll just cancel the init and throw an error, thats safer